### PR TITLE
Run the SFTP test suite again

### PR DESCRIPTION
**Pull request by exarkun**
Fixes: ticket:3240

---
This change is [Reviewable](https://reviewable.io/reviews/tahoe-lafs/tahoe-lafs/647)

### CODE REVIEW

File: newsfragments/3240.installation [hunk 1]
```diff
@@ -0,0 +1 @@
+The SFTP frontend is now contingent on the "sftp" Tahoe-LAFS extra.  To use it, be sure to install at least "tahoe-lafs[sftp]".
```
Comment (habnabit) [inline newsfragments/3240.installation]: Should this have a corresponding documentation update somewhere? I don't know where the tutorial-style prose docs live.

File: src/allmydata/test/test_sftp.py [hunk 2]
```diff
@@ -38,8 +35,8 @@
 class Handler(GridTestMixin, ShouldFailMixin, ReallyEqualMixin, unittest.TestCase):
     """This is a no-network unit test of the SFTPUserHandler and the abstractions it uses."""
 
-    if not have_pycrypto:
-        skip = "SFTP support requires pycrypto, which is not installed"
+    if conch_unavailable_reason:
+        skip = "SFTP support requires Twisted Conch which is not available: {}".format(e)
```
Comment (habnabit) [inline src/allmydata/test/test_sftp.py]: Probably meant `.format(conch_unavailable_reason)` here?

File: newsfragments/3240.installation (hunk 1 above)
Comment (exarkun) [reply newsfragments/3240.installation]: Thanks for that excellent point!  I've added something.

File: src/allmydata/test/test_sftp.py (hunk 2 above)
Comment (exarkun) [reply src/allmydata/test/test_sftp.py]: Woops.  Indeed.

**Pull request merged**